### DISCUSSION TITLE
bn254-mpc setup issue

### DISCUSSION THREAD

**Discussion by cxmqq333** [Consensys/gnark#1231] [Q&A]
when I run the TestSetupCircuit method of bn254/mpcsetup/setup_test.go (master branch or v0.10.0 ）it is ok.
But when I change the circuit to the github Example CubicCircuit and run , there is always a err on call groth16.Prove: 
**Error:      	Received unexpected error:
        	  len(points) != len(scalars)**

![2024-07-30 1](https://github.com/user-attachments/assets/c343c0ef-397c-4d0d-bea8-56a3d3d119fa)
![2024-07-30 2](https://github.com/user-attachments/assets/f336b240-4f9c-4ded-a44e-780d8671678b)
![2024-07-30 3png](https://github.com/user-attachments/assets/b31b7c21-d8ec-4e6a-bc5f-8100756a601b)

**Comment by ivokub** [comment]
It may be a bug - we have open issue for MPC key generation open https://github.com/Consensys/gnark/pull/1180. Do the examples work when the keys are generating using trusted setup (without MPC)?

**Reply by cxmqq333** [reply]
yes, it works when using trusted setup (without MPC). It may be the power is too big for the curcuit.

**Reply by ivokub** [reply]
Alright - I guess we would have to look into MPC. We have been a bit busy with Linea open-sourcing lately, but it is definitely important to resolve.

**Reply by doubiliu** [reply]
> Alright - I guess we would have to look into MPC. We have been a bit busy with Linea open-sourcing lately, but it is definitely important to resolve.

I  have a question. I am using MPC to generate the corresponding Phase1 and Phase2 files and export the corresponding verification contract. But I found that even if I use the same Phase1 and Phase2 files, the exported verification contract will be different.This causes the proof I calculated to be unable to adapt to the contract.

**Reply by ivokub** [reply]
See https://github.com/Consensys/gnark/issues/1449#issuecomment-2746529552
